### ISSUE TITLE
Dorado v0.8.0 declares empty sequence on basecall where v0.7.2 works fine

### ISSUE THREAD

**Issue by tcairns-soundag** [nanoporetech/dorado#1064]
# Issue Report

## Please describe the issue:

We have been using dorado v0.7.2 for some time now and wanted to test results of v0.8.0 before migrating our entire pipeline over. For the same pod5 files dorado v0.7.2 with model v4.2.0 has worked totally fine, but running dorado v0.8.0 with model 4.2.0 or 5.0.0 has produced a runtime error saying an empty sequence is provided for one of the read ids. Basecalling seems to work with dorado v0.8.0 initially (a few hours), but then throws an error with this read. Dorado v0.7.2 on the same machine and with the same pod5 files worked as expected. Is there any reason why v0.7.2 would be able to handle this read but v0.8.0 cannot?

Any insight is greatly appreciated! 

## Steps to reproduce the issue:

Running dorado v0.7.2 and v0.8.0 on the same data on the same machine (at different times) using the same v4.2.0 model.

## Run environment:
- Dorado version: 0.8.0
- Dorado command: `dorado basecaller dorado_models/dna_r10.4.1_e8.2_400bps_sup@v4.2.0 test_dorado_v8_Output/pod5/ --modified-bases 5mC --kit-name SQK-NBD114-24 > test_dorado_v8_Output/doradov0.8.0_modelv4.2.0.bam`
- Operating system: ubuntu linux 6.5.0-44-generic
- Hardware (CPUs, Memory, GPUs): NVIDIA RTX A5000 GPU with 24564MiB of memory, driver version 535.183.01, and CUDA v12.2
- Source data type: pod5s
- Source data location: local on device
- Details about data: 711GB produced on ONT P2 Solo

## Logs

![Screenshot 2024-09-30 at 9 11 43 AM](https://github.com/user-attachments/assets/8a2655c9-0e02-48b3-8b52-390832dffbb6)

**Comment by nallsing-salk** [comment]
Hello,
I am getting the same issue. Running
```
./dorado-0.7.2-linux-x64/bin/dorado basecaller -r --min-qscore 9 --kit-name SQK-RBK114-96 dna_r10.4.1_e8.2_400bps_sup@v5.0.0 pod5/ > dorado-0.7.2.bam
```
successfully completes, while running
```
./dorado-0.8.0-linux-x64/bin/dorado basecaller -r --min-qscore 9 --kit-name SQK-RBK114-96 dna_r10.4.1_e8.2_400bps_sup@v5.0.0 pod5/ > dorado-0.8.0.bam
```
fails with
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Empty sequence and qstring provided for read id 3f0a29db-caff-45bf-953e-84d2cfd1cee7
Aborted (core dumped)
```
I believe this is related to issue #1020 where trimming and/or barcoding does not work properly in the basecall functionality. The replies say we will need to separately basecall with `--no-trim`, then demultiplex. I reran 0.8.0 with
```
./dorado-0.8.0-linux-x64/bin/dorado basecaller -r --min-qscore 9 --no-trim dna_r10.4.1_e8.2_400bps_sup@v5.0.0 pod5/ > dorado-0.8.0.no_trim.bam
./dorado-0.8.0-linux-x64/bin/dorado demux -o demuxed_reads/ --kit-name SQK-RBK114-96 dorado-0.8.0.no_trim.bam
```
and it completed without errors. I think we will have to do this, or use an older version, until they come out with an update.

**Comment by malton-ont** [comment]
This issue should be resolved in dorado 0.8.1, which has just been released.
